src/cow-react/modules/wallet/web3-react/connectors/Injected/index.tsx

```diff
@@ -81,41 +92,26 @@ export class InjectedWallet extends Connector {
 
           throw error
         })
-    }
+      }
+    )
+  }
 
-    return Promise.all([
-      this.provider.request({ method: 'eth_chainId' }) as Promise<string>,
-      this.provider.request({ method: 'eth_requestAccounts' }) as Promise<string[]>,
-    ]).then(([chainId, accounts]) => {
-      const receivedChainId = this.parseChainId(chainId)
+  public async getAccounts() {
+    const { provider } = this
 
-      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-      if (!desiredChainId || desiredChainId === receivedChainId) {
-        return this.actions.update({ chainId: receivedChainId, accounts })
-      }
+    if (!provider) {
+      return []
+    }
 
-      const desiredChainIdHex = `0x${desiredChainId.toString(16)}`
-
-      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-      return this.provider!.request<void>({
-        method: 'wallet_switchEthereumChain',
-        params: [{ chainId: desiredChainIdHex }],
-      }).catch(async (error: ProviderRpcError) => {
-        console.log('err!', error)
-        console.log('debug err', error)
-
-        if (error.code === 4902 && typeof desiredChainIdOrChainParameters !== 'number') {
-          // if we're here, we can try to add a new network
-          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-          return this.provider!.request<void>({
-            method: 'wallet_addEthereumChain',
-            params: [{ ...desiredChainIdOrChainParameters, chainId: desiredChainIdHex }],
-          })
-        }
+    try {
+      const accounts = await provider.request({ method: 'eth_requestAccounts' })
 
-        throw error
-      })
-    })
+      return accounts as string[]
+    } catch (e) {
+      console.log('Failed to get account with eth_requestAccounts method')
+
+      return provider.request({ method: 'eth_accounts' }) as Promise<string[]>
```

Comment:
`Provider` interface has generic type, it should be used instead of type casting

Reply:
Not sure how to do this exactly but if I remove it it give me TS error and they also used it in web3-react.